oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/IndexUpdate.java

```diff
@@ -72,6 +73,9 @@
 
     private static final Logger log = LoggerFactory.getLogger(IndexUpdate.class);
     private static final String TYPE_ELASTICSEARCH = "elasticsearch";
+    //This is used so that wrong index definitions are sparsely logged. After every 1000 indexing cycles with diff, wrong index definitions
```

Comment:
By default the indexing cycle is once every 5 seconds. That mean 5000 seconds, or 83 minutes. I think that's  a good value.

Reply:
And for sync indexes, once in 1000 updates is fine as well I think.